visualization/app/codeCharta/services/loadInitialFile/loadInitialFile.service.ts

```diff
@@ -25,50 +71,351 @@ export class LoadInitialFileService {
+			if (!savedCcState) {
+				this.loadFileService.loadFiles(urlNameDataPairs)
+				this.setRenderStateFromUrl()
+				return
+			}
+
+			const savedFileStates = savedCcState.files
+			const savedNameDataPairs = savedFileStates.map(fileState => getNameDataPair(fileState.file))
+			const urlNameDataPairCheckSums = urlNameDataPairs.map(urlNameDataPair => urlNameDataPair.content.fileChecksum)
+			const savedNameDataPairCheckSums = savedNameDataPairs.map(savedNameDataPair => savedNameDataPair.content.fileChecksum)
+			if (stringify(urlNameDataPairCheckSums) === stringify(savedNameDataPairCheckSums)) {
+				this.applyAppSettings(savedCcState.appSettings)
+				this.loadFileService.loadFiles(savedNameDataPairs)
+				this.store.dispatch(setFiles({ value: savedFileStates }))
+				this.applyAllSettings(savedCcState)
+				this.setRenderStateFromUrl()
+			} else {
+				this.applyAllSettings(savedCcState)
+				this.loadFileService.loadFiles(urlNameDataPairs)
+				this.setRenderStateFromUrl()
+			}
 		} catch (error) {
-			this.tryLoadingSampleFiles(error as Error & { statusText?: string; status?: number })
+			await this.handleErrorLoadFilesFromQueryParams(error as Error)
 		}
 	}
 
-	private tryLoadingSampleFiles(error: Error & { statusText?: string; status?: number }) {
-		if (this.urlUtils.getParameterByName("file")) {
-			const message = "One or more files from the given file URL parameter could not be loaded. Loading sample files instead."
-			let title = "Error"
-			if (error.message) {
-				title += ` (${error.message})`
-			} else if (error.statusText && error.status) {
-				title += ` (${error.status}: ${error.statusText})`
+	private async handleErrorLoadFilesFromQueryParams(error: Error) {
+		if ((error as Error).message !== NO_FILES_LOADED_ERROR_MESSAGE) {
+			const title = "File(s) could not be loaded from the given file URL parameter. Loaded sample files instead."
+			const message = this.createTitleUrlErrorDialog(error as Error)
+			this.showErrorDialog(title, message)
+		}
+		await this.loadSampleFiles()
+	}
+
+	private async loadFilesFromIndexedDB() {
+		try {
+			const savedCcState = await readCcState()
+			if (!savedCcState) {
+				await this.loadSampleFiles()
+				return
 			}
-			this.dialog.open(ErrorDialogComponent, {
-				data: { title, message }
-			})
+
+			const savedFileStates = savedCcState.files
+			const savedNameDataPairs = savedFileStates.map(fileState => getNameDataPair(fileState.file))
+			this.applyAppSettings(savedCcState.appSettings)
+			this.loadFileService.loadFiles(savedNameDataPairs)
+			this.store.dispatch(setFiles({ value: savedFileStates }))
+			this.applyAllSettings(savedCcState)
+		} catch (error) {
+			await this.handleErrorLoadFilesFromIndexedDB(error as Error)
+		}
+	}
+
+	private async handleErrorLoadFilesFromIndexedDB(error: Error) {
+		if ((error as Error).message !== NO_FILES_LOADED_ERROR_MESSAGE) {
+			const title = "File(s) could not be loaded from indexeddb. Loaded sample files instead."
```

Comment:
Can we reframe this a little bit to express that previously loaded files and settings could not be restored?